modules/social_features/social_group/social_group.module

```diff
@@ -1699,8 +1699,43 @@ function social_group_user_cancel_methods_alter(&$methods) {
  * Implements hook_social_user_account_header_create_links().
  *
  * Adds the "Create Group" link to the content creation menu.
+ * When the user can only create groups of one type, the user skips the
+ * group type selection page.
  */
 function social_group_social_user_account_header_create_links($context) {
+  // We can't create a "Add Groups" link without a user to determine permission.
+  if (empty($context['user'])) {
```

Comment:
This is not entirely true because the previous method already worked without a user. In this case we could just default to the normal group type choice page.

Reply:
Agreed.